### PR TITLE
Opacity and zoom for android  rebased for capacitor-3

### PR DESCRIPTION
**Pull request by johnbyrne7**
Added opacity control for web and Android.  Suitable for AR type projects.

Also added pinch/zoom for Android.  This works for both toBack options (true, false),

### CODE REVIEW

**Comment by pbowyer** [comment]
Thanks @johnbyrne7, can you please change the target branch to capacitor-3 as it seems GitHub doesn't give me a way to do this in the UI. Thanks!
image

**Comment by damngamerz** [comment]
@pbowyer Can you merge this and release the new beta? I'm using camera-preview with appflow deployments for development and testing. would be super helpful to have this there, so I can migrate my project to capacitor 3. Thank you.

**Comment by pbowyer** [comment]
@damngamerz I plan to merge it this week 👍

I don't have permissions to release it though, so we have to wait on someone else to do that.

**Comment by arielhernandezmusa** [comment]
@pbowyer let me know when you need to release it, I can do it.

**Comment by pbowyer** [comment]
@arielhernandezmusa Awesome! Could you push out [beta 1](https://github.com/capacitor-community/camera-preview/tree/v2.0.0-beta.1) now please, and later this week I'll finalise beta2 with this and another PR merged?

New Mac should be delivered this week which will make testing _a lot_ quicker 🤩

**Comment by damngamerz** [comment]
@arielhernandezmusa The changes I need for iOS build are present in beta 1. Can you release this? Would be really helpful. Thanks :smile:

**Comment by Raphael-Schulz** [comment]
Hey @arielhernandezmusa our team would also need the changes from [beta 1](https://github.com/capacitor-community/camera-preview/tree/v2.0.0-beta.1). Would be awesome if it could be released soon 😃

**Comment by damngamerz** [comment]
@pbowyer @arielhernandezmusa any updates here?

**Comment by pbowyer** [comment]
@damngamerz I can't do anything to get it released, sorry.

I haven't finished testing this PR yet; did all features work for you? And which version(s) of Android did you test on?

File: src/web.ts [hunk 1]
```diff
@@ -153,6 +154,14 @@ export class CameraPreviewWeb extends WebPlugin implements CameraPreviewPlugin {
   async flip(): Promise<void> {
     throw new Error('flip not supported under the web platform');
   }
+  
+  async setOpacity(_options CameraOpacityOptions): Promise<any> {
```
Comment (pbowyer) [inline src/web.ts:158]:
```suggestion
  async setOpacity(_options: CameraOpacityOptions): Promise<any> {
```
Missing colon

File: src/web.ts [hunk 2]
```diff
@@ -153,6 +154,14 @@ export class CameraPreviewWeb extends WebPlugin implements CameraPreviewPlugin {
...
+    const video = <HTMLVideoElement>document.getElementById("video");
+    if (!!video && !!_options['opacity']) {
+	   video.style.setProperty("opacity", _options['opacity']);
```
Comment (pbowyer) [inline src/web.ts:161]:
```suggestion
	   video.style.setProperty("opacity", _options['opacity'].toString());
```

**Comment by damngamerz** [comment]
@pbowyer Sorry if my message wasn't clear before. I meant it for the [beta 1](https://github.com/capacitor-community/camera-preview/tree/v2.0.0-beta.1), this PR isn't the priority for me atm. We can take time to merge it.

Though I do need beta 1 as it fixes the versions to 12 in iOS which is a minimum requirement for capacitor 3 bump.

**Review by pbowyer — CHANGES_REQUESTED** [review]
Thank you for these two additions - they're awesome!

In the same way opacity can be enabled/disabled, can you add a setting to enable/disable zoom? I couldn't see one already present, but do correct me if that's wrong.

**Comment by pbowyer** [comment]
> Isn't there somebody in the community who you can reach to get access to npm registry?

@damngamerz I have tried :) Rightfully so they keep tight control over who can publish to NPM. I'll see what I can do - NPM is brand new to me but I'm looking to release my own package(s) to side-step the issue.

**Comment by pbowyer** [comment]
@damngamerz @Raphael-Schulz  Please try installing this: https://www.npmjs.com/package/@pbowyer/camera-preview

File: src/web.ts (hunk 1 above)
Comment (johnbyrne7) [reply src/web.ts]: updated PR

File: src/web.ts (hunk 2 above)
Comment (johnbyrne7) [reply src/web.ts]: updated PR

File: android/src/main/java/com/ahm/capacitor/camera/preview/CameraActivity.java [hunk 3]
```diff
@@ -164,75 +166,88 @@ public boolean onTouch(View v, MotionEvent event) {
...
+							// A callback to JS might make sense here.
+						  } else {
+							Log.d(TAG, "onTouch:" + " setFocusArea() did not suceed");
+						  }
+						}
+					  });
+					}
+					return true;
+				  } else {
+					if (dragEnabled) {
```
[the remark this answers is not in the document]
Comment (johnbyrne7) [reply android/src/main/java/com/ahm/capacitor/camera/preview/CameraActivity.java:212]: added enableZoom option in definitions.  defaults to false

**Review by pbowyer — APPROVED** [review]
Sorry for the delay, I wasn't notified you'd pushed!

**Pull request merged**